integrations/extensions/starter-kits/language-model-conversational-search/README.md

```diff
@@ -63,24 +65,117 @@ If this does not suit your needs and you want to experiment with different promp
\ No newline at end of file
+SQuAD (including v2) has very concise answers, which could be one of the reasons why FLAN-UL2 is so terse when using SQuAD v2 prompts.
+
+## Example 2: Connect your assistant to Hugging Face, Milvus, and OpenAI via custom extensions
+
+This example shows how to use Hugging Face to generate query embeddings for semantic search and use those results to generate an answer with the OpenAI model.
+
+Before you upload the sample action for this starter kit, you first need to configure multiple custom extensions: [Hugging Face Hub Embeddings](https://huggingface.co/blog/getting-started-with-embeddings), [Milvus](https://milvus.io) for semantic search, and [OpenAI](../language-model-openai/README.md) for answer generation.
+
+Follow the steps in the following sections for the custom extensions setup before uploading the sample action.
+
+### Configure the Hugging Face Hub Embeddings extension
+
+1. Sign up at [Hugging Face](https://huggingface.co/) and follow the instructions there to get an API key. Save this API key somewhere safe and accessible.
+
+1. Download the OpenAPI specification file [hf-hub-embed.json](hf-hub-embed.json) from this starter kit. You use this specification file to create and add the Hugging Face Hub Embeddings extension to your assistant.
+
+1. In your assistant, on the **Integrations** page, click **Build custom extension** and use the OpenAPI specification file to build a custom extension named `Hugging Face Hub Embeddings`. For general instructions on building any custom extension, see [Building the custom extension](https://cloud.ibm.com/docs/watson-assistant?topic=watson-assistant-build-custom-extension#building-the-custom-extension).
+
+1. After you build the Hugging Face Hub Embeddings extension and it appears on your **Integrations** page, click **Add** to add it to your assistant. For general instructions on adding any custom extension, see [Adding an extension to your assistant](https://cloud.ibm.com/docs/watson-assistant?topic=watson-assistant-add-custom-extension).
+
+1. In **Authentication**, choose **Bearer auth**. Copy and paste your Hugging Face API key into the **Token** field.
+
+1. The setup of your extension is now complete. For more information about the extension parameters, see the [Hugging Face API Reference](https://huggingface.co/docs/hub/api).
+
+### Configure the Milvus semantic search extension
+
+Milvus is an open-source vector database for semantic search. In this example we use Milvus to search a document collection that we previously indexed with Milvus. [The python script to index documents for search](index-with-milvus.py) is included for reference. In our example we use `.pdf` documents; the script can be modified to use other text formats.
+
+1. Sign up at [Milvus](https://milvus.io) and follow the instructions there to get access to the Milvus server variables that you need to connect this extension with your assistant.
```

Comment:
This should say to install Milvus somewhere that WA can access.  Install instructions are at https://milvus.io/docs/install_standalone-docker.md .  One option for where to host is an IBM virtual server: https://www.ibm.com/cloud/virtual-servers .  Another is IBM code engine: https://www.ibm.com/cloud/code-engine .  However, users can also install on a machine they own as long as it is connected to the internet.  The could also run Milvus on a different cloud such as AWS.  Users then need to record the IP address of the machine where they installed Milvus and the port where it is running.